src/repository.rs

```diff
@@ -1,23 +1,58 @@
 //! Repositories are abstraction over a specific mongo collection for a given `Model`
 
 use crate::{CollectionConfig, Model};
-use mongodb::bson::{from_bson, to_bson, Bson, Document};
+use mongodb::bson::oid::ObjectId;
+use mongodb::bson::{doc, from_bson, to_bson, Bson, Document};
 use mongodb::options::*;
 use mongodb::results::*;
+use serde::Deserialize;
+use std::borrow::Borrow;
+
+#[derive(Debug)]
+pub struct BulkUpdate {
+    pub query: Document,
+    pub update: Document,
+    pub options: Option<UpdateOptions>,
+}
+
+impl AsRef<BulkUpdate> for BulkUpdate {
+    fn as_ref(&self) -> &BulkUpdate {
+        &self
+    }
+}
+
```

Comment:
I don't think it makes much sense to implement `AsRef` on `BulkUpdate` as it doesn't sound like we'll ever want to do reference conversions.

```suggestion
```

Reply:
True, I initially took AsRef in the method instead of Borrow, but now this isn't needed anymore